File: Creational/Builder/Python/Structure/concrete_builder.py
from builder import Builder
from product import Product


class ConcreteBuilder1(Builder):
    """
    The Concrete Builder classes follow the Builder interface and provide
    specific implementations of the building steps. Your program may have several
    variations of Builders, implemented differently.
    """
    
    def __init__(self):
        self.reset()
    
    def reset(self):
        """Reset the builder to create a new product"""
        self._product = Product()
    
    def build_part_a(self):
        """Build part A for ConcreteBuilder1"""
        self._product.add_part("PartA1")
        return self
    
    def build_part_b(self):
        """Build part B for ConcreteBuilder1"""
        self._product.add_part("PartB1")
        return self
    
    def build_part_c(self):
        """Build part C for ConcreteBuilder1"""
        self._product.add_part("PartC1")
        return self
    
    def get_result(self):
        """Return the built product"""
        product = self._product
        self.reset()
        return product


class ConcreteBuilder2(Builder):
    """
    The Concrete Builder classes follow the Builder interface and provide
    specific implementations of the building steps. Your program may have several
    variations of Builders, implemented differently.
    """
    
    def __init__(self):
        self.reset()
    
    def reset(self):
        """Reset the builder to create a new product"""
        self._product = Product()
    
    def build_part_a(self):
        """Build part A for ConcreteBuilder2"""
        self._product.add_part("PartA2")
        return self
    
    def build_part_b(self):
        """Build part B for ConcreteBuilder2"""
        self._product.add_part("PartB2")
        return self
    
    def build_part_c(self):
        """Build part C for ConcreteBuilder2"""
        self._product.add_part("PartC2")
        return self
    
    def get_result(self):
        """Return the built product"""
        product = self._product
        self.reset()
        return product

File: Creational/Builder/Php/Structure/ConcreteBuilder.php
<?php

declare(strict_types=1);

require_once __DIR__ . '/Product.php';
require_once __DIR__ . '/Builder.php';

class ConcreteBuilder implements Builder
{
    private Product $product;

    public function __construct()
    {
        $this->reset();
    }

    public function reset(): void
    {
        $this->product = new Product();
    }

    public function buildPartA(): void
    {
        $this->product->addPart('PartA');
    }

    public function buildPartB(): void
    {
        $this->product->addPart('PartB');
    }

    public function buildPartC(): void
    {
        $this->product->addPart('PartC');
    }

    public function getProduct(): Product
    {
        $result = $this->product;
        $this->reset();
        return $result;
    }
}


